deploy/determined_deploy/gcp/gcp.py

```diff
@@ -261,3 +267,30 @@ def delete(configs: Dict, env: Dict) -> None:
 
 def run_command(command: str, env: Dict[str, str]) -> None:
     subprocess.check_call(command, env=env, shell=True, stdout=sys.stdout)
+
+
+def validate_gcp_credentials(configs: Dict) -> None:
+    vars_file_path = get_terraform_vars_file_path(configs)
+    tf_vars = terraform_read_variables(vars_file_path)
+    set_gcp_credentials_env(tf_vars)
+
+    try:
+        googleapiclient.discovery.build("compute", "v1")
```

Comment:
if we wanted to get even more fancy one day, it looks like we can test the permissions associated with an IAM account to make sure they match what we need https://cloud.google.com/iam/docs/testing-permissions#iam-testing-permissions-python

Reply:
yeah - I suppose it'll need to include both this (what we do from python code) and whatever resources terraform requires.